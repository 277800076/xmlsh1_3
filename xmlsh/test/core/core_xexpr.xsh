# Test XML Expressions

A=value
B=attribute
# turn off indentation for this test
set +indent

# empty sequence
echo empty sequence <[()]>

# simple mixed sequence
echo <[1,2,"string",<foo>bar</foo>]>

# constructed element 
#
echo the following crashes saxon sometimes
X=<[<foo a1='{$B}'>{$A}</foo>]>
#X=<[<foo>{$A}</foo>]>
echo <[<bar>{$X}</bar>]>

# for loop with sequences
# each xexpr is treated as 1 expression
for e in <[1,"string",<simple/>,$X]> <[2,3,4]>; do
   echo for: $e
done

# Positional paramters as expressions
set $X <["second","arg"]>

echo Number of params: $#
if [ $# -ne 2 ] ; then 
   echo Wrong number of positional parameters ;
   exit 1
fi

echo $1
echo $2

exit 0



