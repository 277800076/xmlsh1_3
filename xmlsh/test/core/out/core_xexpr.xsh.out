empty sequence
1 2 string <foo>bar</foo>
the following crashes saxon sometimes
<bar>
   <foo a1="attribute">value</foo>
</bar>
for: 1 string <simple/> <foo a1="attribute">value</foo>
for: 2 3 4
Number of params: 2
<foo a1="attribute">value</foo>
second arg
